### PR TITLE
[INLONG-10173][Sort] SortStandalone support request unified configuration

### LINKED ISSUE #10173: [Feature][Sort] SortStandalone support request unified configuration
### Description

parent #10164

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
Fixes #10173

### Motivation

To support the unified sort configuration

### Modifications

The configuration request logic

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/v2/SortTask.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    }
+
+    public void start() {
+        SortTaskConfig config = SortClusterConfigHolder.getTaskConfig(taskName);
+        if (config == null) {
+            return;
+        }
+        //
+        Map<String, String> flumeConfiguration = FlumeConfigGenerator.generateFlumeConfiguration(config);
+        log.info("Start sort task:{},config:{}", taskName, flumeConfiguration);
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/v2/SortTask.java:57]: Start sort task={}, config={}

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/v2/SortCluster.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+                    this.deletingTasks.add(entry.getValue());
+                }
+            }
+            // stop deleting task list
+            for (SortTask task : deletingTasks) {
+                task.stop();
+                taskMap.remove(task.getTaskName());
+            }
+        } catch (Throwable e) {
+            log.error("failed to reload cluster", e);
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/v2/SortCluster.java:140]: log.error("Failed to reload cluster", e);

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/v2/SortCluster.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            // stop sort task
+            for (Map.Entry<String, SortTask> entry : this.taskMap.entrySet()) {
+                entry.getValue().stop();
+            }
+            // stop admin task
+            if (this.adminTask != null) {
+                this.adminTask.stop();
+            }
+        } catch (Exception e) {
+            log.error("failed to close sort cluster", e);
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/v2/SortCluster.java:77]: log.error("Failed to close sort cluster", e);

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/v2/SortCluster.java [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+     */
+    public void start() {
+        try {
+            this.reload();
+            this.setReloadTimer();
+            // start admin task
+            this.adminTask = new AdminTask(new Context(CommonPropertiesHolder.get()));
+            this.adminTask.start();
+        } catch (Exception e) {
+            log.error("failed to start sort cluster", e);
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/v2/SortCluster.java:58]: Ditto.

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/loader/v2/ClassResourceSortClusterConfigLoader.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            if (context != null) {
+                fileName = context.getString(SortClusterConfigType.KEY_FILE, SortClusterConfigType.DEFAULT_FILE);
+            }
+            String confString = IOUtils.toString(getClass().getClassLoader().getResource(fileName),
+                    Charset.defaultCharset());
+            int index = confString.indexOf('{');
+            confString = confString.substring(index);
+            return objectMapper.readValue(confString, SortConfig.class);
+        } catch (Exception e) {
+            LOG.error("fail to load properties, file ={}, and e= {}", fileName, e);
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/loader/v2/ClassResourceSortClusterConfigLoader.java:60]: LOG.error("Failed to load properties, file ={}", fileName, e);

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/v2/SortClusterConfigHolder.java [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            }
+            if (instance.loader == null) {
+                instance.loader = new ClassResourceSortClusterConfigLoader();
+            }
+            try {
+                instance.loader.configure(new Context(CommonPropertiesHolder.get()));
+                instance.reload();
+                instance.setReloadTimer();
+            } catch (Exception e) {
+                log.error("failed to reload instance", e);
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/v2/SortClusterConfigHolder.java:88]: Ditto.

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/v2/SortClusterConfigHolder.java [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+            } else {
+                // user-defined
+                try {
+                    Class<?> loaderClass = ClassUtils.getClass(loaderType);
+                    Object loaderObject = loaderClass.getDeclaredConstructor().newInstance();
+                    if (loaderObject instanceof SortConfigLoader) {
+                        instance.loader = (SortConfigLoader) loaderObject;
+                    }
+                } catch (Throwable t) {
+                    log.error("fail to init loader,loaderType:{},error:{}", loaderType, t.getMessage());
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/v2/SortClusterConfigHolder.java:77]: Ditto.

File: inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/v2/SortClusterConfigHolder.java [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+    }
+
+    private static SortClusterConfigHolder get() {
+        if (instance != null) {
+            return instance;
+        }
+
+        synchronized (SortClusterConfigHolder.class) {
+            instance = new SortClusterConfigHolder();
```
Comment (aloyszhang) [inline inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/v2/SortClusterConfigHolder.java:59]: null check after `synchronized` is needed for singleton

Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-common/src/main/java/org/apache/inlong/sort/standalone/config/holder/v2/SortClusterConfigHolder.java]: fixed, thx

**Review by aloyszhang — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**